build/helper/codegen_helper.py

```diff
@@ -37,21 +45,29 @@
         'skip_self': True,
         'name_to_use': 'python_name_with_doc_default',
     },
-    ParameterUsageOptions.CTYPES_CALL: {
-        'skip_self': True,
+    ParameterUsageOptions.LIBRARY_METHOD_DECLARATION: {
+        'skip_self': False,
         'name_to_use': 'python_name',
     },
     ParameterUsageOptions.LIBRARY_METHOD_CALL: {
+        'skip_self': True,
+        'name_to_use': 'ctypes_method_call_snippet',
```

Comment:
Although, I've looked at the code this is used to generate (a bunch of ctypes parameters), I feel like this key name doesn't fit in with the ENUM's description:
`'''For calling into a Library method (from LibraryInterpreter).'''`.

Reply:
I think I prefer the accuracy of 'ctypes_method_call_snippet' here.

However, after you brought it up, now I'm wondering if maybe something like LazyCDLL would be a more appropriate name than Library, since it's really just a CDLL wrapper that lazy-loads the functions. Doing so would also have the side effect of making 'ctypes_method_call_snippet' make more sense.

Reply:
I don't have any strong opinions on the name of the Library class. I'm fine if we just leave this as is. If we do rename the Library class, that may trigger quite a few renames, but I'm prepared to review that, if necessary.

Reply:
Improving naming and increasing consistency in how we name things is totally worth the ROI in my opinion. It makes it easier to reason about the code and onboard new developers.

Reply:
Would you prefer the LazyCDLL name, then?

Reply:
I'd rather rename `ctypes_method_call_snippet`.
Why? We have a lot of internal driver code that generates equivalent classes in C++ called `Library`. We are likely to do something similar for C#. It's nice to have that consistency.

Reply:
Somewhere we have to make the connection that "when you call into Library, you build and pass ctypes parameters." If we don't do it here, then that connection will then instead be implied in metadata_add_all.py, where we'd fill in the library_method_call_snippet (renamed from ctypes_method_call_snippet) with ctypes-specific info.

Personally, I think that right here in codegen_helper.py's dictionary mapping is the ideal place to connect the two concepts, and let metadata_add_all deal with the lower-level concept of "you want to pass ctypes? here's how."

Reply:
I could go either way on this.
I think if we keep `'name_to_use': 'ctypes_method_call_snippet',`, though, you should add a comment `# when you call into Library, you build and pass ctypes parameters`.

As it is, it looks wrong because of how name_to_use matches up in all of the other METHOD_CALL objects.

Reply:
The main purpose of the comment would be to acknowledge that the _method_call_snippet name doesn't match and make it clear that it's not a mistake.

Reply:
Added comment